src/decoder/mod.rs

```diff
@@ -390,8 +389,10 @@ impl<R: Read> Reader<R> {
             }
         }
 
+        let mut buf = Vec::new();
```

Comment:
I believe it might be possible to just use self.prev here as well instead of allocating a new buffer.